### PR TITLE
Automated task for validating rotation of etcd encryption keys

### PR DESCRIPTION
**Pull request by SurajGudaji**

### CODE REVIEW

**Review by varad-ahirwadkar — COMMENTED** [review]
Can we use reuse this role - https://github.com/ocp-power-automation/ocp4-playbooks-extras/tree/main/playbooks/roles/ocp-etcd-encryption-decryption ?

**Comment by SurajGudaji** [comment]
> Can we use reuse this role - https://github.com/ocp-power-automation/ocp4-playbooks-extras/tree/main/playbooks/roles/ocp-etcd-encryption-decryption ?

we can but this role is defined very statically, so to use it we will need to make some changes to it
i.e calling the the encryption or decryption tasks based on the variables passed by ocp-etcd-rotate-encryption-key role

File: playbooks/roles/ocp-etcd-rotate-encryption-key/tasks/main.yaml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    container: etcd
+    local_path: "{{ role_path }}/files/secret.sh"
+    remote_path: /tmp/secret.sh
+    state: to_pod
+
+- name: Increment NUMBER variable
+  set_fact:
+    NUMBER: "{{ NUMBER + 1 }}"
+
+- name: Execute secret collecter script in etcd pod to collect encrypted value of secrets after key rotation
```
[the remark this answers is not in the document]
Comment (SurajGudaji) [reply playbooks/roles/ocp-etcd-rotate-encryption-key/tasks/main.yaml:77]: yes we can

**Comment by varad-ahirwadkar** [comment]
> we can but this role is defined very statically, so to use it we will need to make some changes to it i.e calling the the encryption or decryption tasks based on the variables passed by ocp-etcd-rotate-encryption-key role

Are you planning to implement these changes?

**Comment by SurajGudaji** [comment]
> > we can but this role is defined very statically, so to use it we will need to make some changes to it i.e calling the the encryption or decryption tasks based on the variables passed by ocp-etcd-rotate-encryption-key role
> 
> Are you planning to implement these changes?

I was planning to do those changes in next PR because, some more changes related to validation of etcd encryption task were in line, so I will make these changes in next PR

**Comment by varad-ahirwadkar** [comment]
@pravin-dsilva can you please review this PR? Thanks

**Review by yussufsh — CHANGES_REQUESTED** [review]

**Review by yussufsh — APPROVED** [review]
/lgtm

**Pull request merged**